### PR TITLE
Case 20138: Fix vive crash (RC 76)

### PR DESCRIPTION
**Pull request by SamGondelman**
https://github.com/highfidelity/hifi/pull/14534

### CODE REVIEW

File: plugins/openvr/src/OpenVrDisplayPlugin.cpp [hunk 1]
```diff
@@ -485,6 +485,7 @@ bool OpenVrDisplayPlugin::internalActivate() {
                 _submitCanvas->doneCurrent();
             });
         }
+        _submitCanvas->moveToThread(_submitThread.get());
```
Comment (jherico) [inline plugins/openvr/src/OpenVrDisplayPlugin.cpp]: needs to be inside the if block

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15661/HighFidelity-Beta-PR14535-e7e2b0b.apk

File: plugins/openvr/src/OpenVrDisplayPlugin.cpp (hunk 1 above)
Comment (SamGondelman) [reply plugins/openvr/src/OpenVrDisplayPlugin.cpp]: I don't think so.  This is in `internalActivate`.  We make a new _submitThread every time, but keep the same _submitCanvas.  The canvas always need to be moved to the new thread.

Comment (jherico) [reply plugins/openvr/src/OpenVrDisplayPlugin.cpp]: If that's the case you'll also need code in the submit thread to move the canvas back to the main thread when the submit thread dies.  

Basically you need to test locally with turning the plugin on and off several times.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15661/HighFidelity-Beta-PR14535-e7e2b0b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15661/HighFidelity-Beta-Interface-PR14535-e7e2b0b.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15664/HighFidelity-Beta-PR14535-98d3f6a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15664/HighFidelity-Beta-PR14535-98d3f6a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15664/HighFidelity-Beta-Interface-PR14535-98d3f6a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15664/HighFidelity-Beta-PR14535-98d3f6a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15664/HighFidelity-Beta-Interface-PR14535-98d3f6a.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass
No crash when switching to Vive.

**Pull request merged**